src/apps/service/apps/de/listings.clj

```diff
@@ -145,19 +145,24 @@
 
 (def ^:private virtual-group-ids (set (keys virtual-group-fns)))
 
+(defn- realize-group
+  [group]
+  (reduce (fn [group k] (if (or (future? (group k)) (delay? (group k))) (update group k deref) group)) group (keys group)))
+
 (defn- format-private-virtual-groups
   "Formats any virtual groups that should appear in a user's workspace."
   [user workspace params]
-  (remove :is_public
-          (map (fn [[_ {f :format-group}]] (f user workspace params)) virtual-group-fns)))
+  (map realize-group
+    (doall (remove :is_public ;; resolve immediately to start futures executing
+      (map (fn [[_ {f :format-group}]] (f user workspace params)) virtual-group-fns)))))
 
 (defn- add-private-virtual-groups
   [user group workspace params]
   (let [virtual-groups (format-private-virtual-groups user workspace params)
-        actual-count   (count-apps-in-group-for-user
-                        (:id group)
-                        (:username user)
-                        params)]
+        actual-count   (future (count-apps-in-group-for-user
+                                 (:id group)
+                                 (:username user)
+                                 params))]
```

Comment:
I realized this could be made a future too, because nothing depends on its value until it's returned to the client.

Reply:
(to be clear here, the prewalks ensure this is made into a real value before returning to the client. Assuming I didn't miss any, anyway)